### PR TITLE
Adding automatic channel name to Hex color mapping

### PR DESCRIPTION
**Pull request by edyoshikun**
Here I made a dictionary with the most popular colors and most of the channels utilized in our imaging platforms including the label-free channels `(i.e S0,S1,S4, Retardance,Phase,etc)` and fluorophores `(i.e GFP, TXR, DAPI, etc)` to automate the channel name to hex color mapping #83 .

For the color mapping, I have the red fluorophores mapped to magenta as magenta-green is preferred for visualization to be inclusive for folks with deuteranomaly. We currently don't have an easy way to handle HSV #74  except splitting the HSV into RGB, in which each channel is named `red`, `blue`, and `green`.  

Screenshot 2023-03-29 at 2 59 18 PM

### CODE REVIEW

File: examples/single_fov_ome_zarr.py [hunk 1]
```diff
@@ -11,22 +11,23 @@
...
 
 from iohub.ngff import open_ome_zarr
+from datetime import datetime
 
 # %%
 # Set storage path
-
-store_path = f"{tempfile.gettempdir()}/ome.zarr"
+timestamp = datetime.now().strftime('%d%H%M%S')
+store_path = '/home/eduardo.hirata/Documents/tmp/' +timestamp+ "ome.zarr"
```
Comment (mattersoflight) [inline examples/single_fov_ome_zarr.py:19]: We shouldn't hardcode any specific path in the repo. I suggest `os.path.expanduser('~/tmp/...')`.

File: iohub/display_utils.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    Returns
+    -------
+    str: Hex string corresponding to the RGB value
+    """
+    hex_color = "{:02X}{:02X}{:02X}".format(r, g, b)
+    return hex_color
+
+# Dictionary with key works and most popular fluorescent probes
+channel_colors = {
+    "lime": ["GFP", "Green", "Alexa488", "GCaMP","FITC", "mNeon"],
```
Comment (mattersoflight) [inline iohub/display_utils.py:30]: green and magenta complement each other, i.e., their overlay is white. Why choose slightly lighter lime?

**Review by mattersoflight — COMMENTED** [review]
I think #74 requires more detailed discussion and should be handled in a separate PR.

File: examples/single_fov_ome_zarr.py (hunk 1 above)
Comment (edyoshikun) [reply examples/single_fov_ome_zarr.py:19]: ups didn't mean to push and alter the example code. Ill remove this one. Good catch.

File: iohub/display_utils.py (hunk 2 above)
Comment (edyoshikun) [reply iohub/display_utils.py:30]: This is also a good point. Initially, I had a typo in my green set as (0,128,0) which made it darker than lime green. Fixed now. Thank you!

File: iohub/display_utils.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
+# Utility functions for displaying data
```
Comment (ziw-liu) [inline iohub/display_utils.py:1]: Module docstrings should be string literals. See [PEP 257](https://peps.python.org/pep-0257/).

File: iohub/display_utils.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+from typing import Tuple
+
+import numpy as np
+
+from iohub.ngff_meta import ChannelMeta, WindowDict
+
+
+def rgb_to_hex(r: int, g: int, b: int) -> str:
```
Comment (ziw-liu) [inline iohub/display_utils.py:10]: iohub stores color metadata as a pydantic color field which converts different color representations into hex:
https://github.com/czbiohub/iohub/blob/866e6690991fee623a00b8c60a3602a4d7b30922/iohub/ngff_meta.py#L247

So this function is likely redundant.

File: iohub/display_utils.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    Returns
+    -------
+    str: Hex string corresponding to the RGB value
+    """
+    hex_color = "{:02X}{:02X}{:02X}".format(r, g, b)
+    return hex_color
+
+
+# Dictionary with key works and most popular fluorescent probes
+channel_colors = {
```
Comment (ziw-liu) [inline iohub/display_utils.py:31]: Can we have this global constant in upper case?

File: iohub/display_utils.py [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+            clim = channel_settings["Other"]
+            display_color = "FFFFFF"
+
+    # Mapping channel name to color
+    for key in channel_colors:
+        if chan_name in channel_colors[key]:
+            display_color = popular_colors[key]
+            break
+        else:
+            display_color = display_color = "FFFFFF"
```
Comment (ziw-liu) [inline iohub/display_utils.py:138]:
```suggestion
            display_color = POPULAR_COLORS["white"]
```

File: iohub/display_utils.py [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+        "Orientation",
+        "State0",
+        "State1",
+        "State2",
+        "State3",
+        "State4",
+    ],
+}
+
+popular_colors = {
```
Comment (ziw-liu) [inline iohub/display_utils.py:65]:
```suggestion
POPULAR_COLORS = {
```

File: iohub/display_utils.py (hunk 5 above)
Comment (ziw-liu) [reply iohub/display_utils.py:31]:
```suggestion
CHANNEL_COLORS = {
```

File: iohub/display_utils.py [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+        "S2": (-0.5, 0.5, -10.0, 10.0),
+        "S3": (-1.0, 1.0, -10.0, -10.0),
+        "Other": (0, U16_FMAX, 0.0, U16_FMAX),
+    }
+    if not clim:
+        if chan_name in channel_settings.keys():
+            clim = channel_settings[chan_name]
+        else:
+            clim = channel_settings["Other"]
+            display_color = "FFFFFF"
```
Comment (ziw-liu) [inline iohub/display_utils.py:130]:
```suggestion

```

**Comment by edyoshikun** [comment]
@mattersoflight 
One interesting thing to note is that in the from the [CSS3 Specifications](https://www.w3.org/TR/css-color-3/#svg-color) `lime` is (0,255,0) and `green` is just an opaque lime (0,128,0). 

Screenshot 2023-03-30 at 8 47 48 AM

From these suggestions, we realized that the current `pydantic.color.Color()` converts the long hex to short hex (i.e #FFFFFF to #fff). However,  the following function was also not working when modifying the color field and calling `.json()`, so I'm getting rid of it since it doesn't work and it also outputs the wrong hex format.
https://github.com/czbiohub/iohub/blob/866e6690991fee623a00b8c60a3602a4d7b30922/iohub/ngff_meta.py#L254

Instead, we will import the color dictionary from the CSS Specs that already exist in PIL. 

Thank you @ziw-liu!

**Comment by edyoshikun** [comment]
Adding to maybe why the json_encoder didn't work is linked [here](https://stackoverflow.com/questions/73168575/pydantic-color-type-hex-only)

Given that it's used here as well. https://github.com/czbiohub/iohub/blob/866e6690991fee623a00b8c60a3602a4d7b30922/iohub/ngff_meta.py#L303

File: iohub/display_utils.py [hunk 9]
```diff
@@ -1,11 +1,47 @@
-# Utility functions for label-free microscopy data
+""" Utility functions for displaying data """
 
 from typing import Tuple
 
 import numpy as np
+from PIL.ImageColor import colormap
```
Comment (ziw-liu) [inline iohub/display_utils.py:6]: Maybe also explicitly depend on `pillow` in `setup.cfg`?

**Comment by ziw-liu** [comment]
> Adding to maybe why the json_encoder didn't work is linked [here](https://stackoverflow.com/questions/73168575/pydantic-color-type-hex-only)
> 
> Given that it's used here as well.
> 
> https://github.com/czbiohub/iohub/blob/866e6690991fee623a00b8c60a3602a4d7b30922/iohub/ngff_meta.py#L303

Thanks for pointing this out! I think I fixed the remaining one in https://github.com/czbiohub/iohub/pull/101/commits/411a51350d3d35fdde28e53e021c6697e6995154.

File: iohub/display_utils.py (hunk 9 above)
Comment (edyoshikun) [reply iohub/display_utils.py:6]: added dependency to current instal version and beyond >= 9.4.0. Is that good?

File: iohub/display_utils.py [hunk 10]
```diff
@@ -50,11 +103,21 @@ def channel_display_settings(
             clim = channel_settings[chan_name]
         else:
             clim = channel_settings["Other"]
+            display_color = color_to_hex("white")
```
Comment (ziw-liu) [inline iohub/display_utils.py:106]:
```suggestion

```
This line seems to be redundant given [L113-114.](https://github.com/czbiohub/iohub/blob/6dec2b247fe9f3468d638513d3a895fde2100cf4/iohub/display_utils.py#L114)

**Review by ziw-liu — APPROVED** [review]
Great first PR!

**Pull request merged**